### PR TITLE
Updated setup script for purdue tests with deeptau

### PR DESCRIPTION
**Pull request by wpmccormack**
Adds in deeptau PR from Yongbin + model files.  A few files need to be retrieved from github right now, so this script will need to be updated in the future.  I tried doing a merge-topic with the actual tau-pog PR branch, but that pulls in basically all of cmssw rather than just a few directories.  The difference between YB's branch and the actual PR branch is the edits in allSonicTriton, and in the runTauIdMVA, because the directory of the model files had to be changed

### CODE REVIEW

**Comment by kpedro88** [comment]
The tau-pog branch is based on a later IB, not pre4, which is why it can't easily be merged. I suggest @yongbinfeng prepares an updated version of his branch based on pre4 but with the tau-pog changes, rather than wgetting things.

**Comment by kpedro88** [comment]
also, how many packages are pulled in by checkdeps based on python dependencies (which are usually pointless)? sometimes it's better to do `git cms-merge-topic -u` and then use `git cms-checkdeps ...` afterward with the correct arguments to pull in only necessary dependencies.

File: setup.sh [hunk 1]
```diff
@@ -90,7 +90,20 @@ cd ${CMSSWVER}/src
...
 git cms-merge-topic fastmachinelearning:CMSSW_13_3_0_pre4_PTTC
+git cms-merge-topic wpmccormack:CMSSW_13_3_0_pre4_new_PN_and_HiggsIN
+git cms-merge-topic yongbinfeng:CMSSW_13_3_0_pre3_DeepTauIdSONIC
 git cms-addpkg RecoBTag/Combined
-git clone ${ACCESS_GITHUB}fastmachinelearning/RecoBTag-Combined -b onnx_patch RecoBTag/Combined/data
+mkdir RecoTauTag/TrainingFiles
+git clone ${ACCESS_GITHUB}wpmccormack/RecoBTag-Combined.git -b onnx_patch_with_newPN_HiggsIN RecoBTag/Combined/data
+git clone ${ACCESS_GITHUB}yongbinfeng/RecoTauTag-TrainingFiles.git -b DeepTau2018v2p5_SONIC RecoTauTag/TrainingFiles/data
 git clone ${ACCESS_GITHUB}fastmachinelearning/sonic-workflows -b CMSSW_13_3_X
+SRCDIR=`pwd`
```
Comment (kpedro88) [inline setup.sh]: better defined as `$CMSSW_BASE/src`

Comment (wpmccormack) [reply setup.sh]: changed over to that

**Comment by wpmccormack** [comment]
I need to check on the checkdeps thing.  I've never used that before

**Comment by wpmccormack** [comment]
> also, how many packages are pulled in by checkdeps based on python dependencies (which are usually pointless)? sometimes it's better to do `git cms-merge-topic -u` and then use `git cms-checkdeps ...` afterward with the correct arguments to pull in only necessary dependencies.

Hi Kevin, I'm a bit confused by this question.  Are you suggesting a change to the setup script?  If I understand this tutorial, https://github.com/cms-sw/cmssdt-wiki/blob/master/cmssw/old/git-cms-addpkg.md, `checkdeps` is used to check the dependencies of any development work you've done locally after adding some packages.  The current `cms-merge-topic` commands that I run in the setup script only bring in a few packages.  After I bring in your PTTC branch, my PN stuff, and YB's DT stuff, checkdeps says there are 4 packages to check out and compile.  If I run `cms-merge-topic` with the cms-tau-pog branch though, checkdeps says there are 883 packages to check out and compile.

**Comment by kpedro88** [comment]
> After I bring in your PTTC branch, my PN stuff, and YB's DT stuff, checkdeps says there are 4 packages to check out and compile.

This is what I wanted to know. Which ones are they?

`cms-merge-topic` automatically calls `cms-checkdeps` unless the `-u` option is used (as I suggested above). `cms-checkdeps` can also be called on its own, and it has flags to determine which kinds of dependencies it checks.

**Comment by wpmccormack** [comment]
The packages that checkdeps gives are
PhysicsTools/NanoAOD
PhysicsTools/PatAlgos
RecoBTag/Configuration
RecoTauTag/Configuration

Do you want me to switch to the -u flag and then add these by hand?

**Comment by kpedro88** [comment]
checkdeps should say the type of dependency (header, python, buildfile). if some of these are python, we can probably omit them.

**Comment by wpmccormack** [comment]
Ah, they're all python.  So I'll switch all the merge-topic commands to -u and then not pull any of the extra dependencies

**Comment by wpmccormack** [comment]
Updated the script, and once Yongbin updates his branch, I'll take out the wget commands

**Comment by yongbinfeng** [comment]
> Updated the script, and once Yongbin updates his branch, I'll take out the wget commands

Done. PRed directly https://github.com/wpmccormack/sonic-workflows/pull/1

**Comment by wpmccormack** [comment]
Awesome, thanks Yongbin!

**Pull request merged**